### PR TITLE
Add const non-null asserts where required

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: packages/flutter/lib/src/gestures/velocity_tracker.dart [hunk 1]
```diff
@@ -87,12 +87,17 @@ class Velocity {
...
   const VelocityEstimate({
     this.pixelsPerSecond,
     this.confidence,
     this.duration,
     this.offset,
-  });
+  }) : assert(pixelsPerSecond != null),
+       assert(confidence != null),
+       assert(duration != null),
+       assert(offset != null);
```
Comment (cbracken) [inline packages/flutter/lib/src/gestures/velocity_tracker.dart]: It looks like this is the intent, but not 100% sure.

File: packages/flutter/lib/src/material/list_tile.dart [hunk 2]
```diff
@@ -173,6 +173,7 @@ class ListTile extends StatelessWidget {
   }) : assert(isThreeLine != null),
        assert(enabled != null),
        assert(selected != null),
+       assert(!isThreeLine || subtitle != null),
```
Comment (cbracken) [inline packages/flutter/lib/src/material/list_tile.dart]: FYI -- this one is special

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 3]
```diff
@@ -993,7 +993,7 @@ class LinearGradient extends Gradient {
 ///  * [CustomPainter], which shows how to use the above sample code in a custom
 ///    painter.
 class RadialGradient extends Gradient {
-  /// Creates a radial graident.
+  /// Creates a radial gradient.
```
Comment (cbracken) [inline packages/flutter/lib/src/painting/box_painter.dart]: Technically not an assert...

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 4]
```diff
@@ -1003,7 +1003,10 @@ class RadialGradient extends Gradient {
     this.colors,
     this.stops,
     this.tileMode: TileMode.clamp,
-  });
+  }) : assert(center != null),
+       assert(radius != null),
+       assert(colors != null),
```
Comment (cbracken) [inline packages/flutter/lib/src/painting/box_painter.dart]: Worth asserting `colors.isNotEmpty` too?

File: packages/flutter/lib/src/material/theme_data.dart [hunk 5]
```diff
@@ -446,6 +446,7 @@ class ThemeData {
   static ThemeData lerp(ThemeData begin, ThemeData end, double t) {
     assert(begin != null);
     assert(end != null);
+    assert(t != null);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: if you're gonna do this one, you should add it to all the `lerp` functions across the framework (and maybe even those in the engine), as well as the `lerp` methods on the Tween subclasses. Maybe do that in a separate PR though...

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/theme_data.dart (hunk 5 above)
Comment (cbracken) [inline packages/flutter/lib/src/material/theme_data.dart]: Pulled this line. Will do the whole lot in one go.
